cli/cli-client/src/main/java/org/infinispan/cli/commands/CLI.java

```diff
@@ -0,0 +1,257 @@
+public class CLI extends CliCommand {
+   public static final String CMD = "cli";
+
+   private Context context;
+
+   @Option(completer = FileOptionCompleter.class, shortName = 't', name = "truststore", description = "A truststore to use when connecting to SSL/TLS-enabled servers")
+   Resource truststore;
+
+   @Option(shortName = 's', name = "truststore-password", description = "The password for the truststore")
+   String truststorePassword;
+
+   @Option(shortName = 'v', hasValue = false, description = "Shows version information")
+   boolean version;
+
+   @Option(hasValue = false, description = "Whether to trust all certificates", name = "trustall")
+   boolean trustAll;
+
+   @Option(completer = FileOptionCompleter.class, shortName = 'f', description = "File for batch mode")
+   Resource file;
+
+   @Option(shortName = 'c', description = "A connection URL. Use '-' to connect to http://localhost:11222")
+   String connect;
+
+   @Option(shortName = 'h', hasValue = false, overrideRequired = true)
+   protected boolean help;
+
+   @Override
+   public boolean isHelp() {
+      return help;
+   }
+
+   @Override
+   public CommandResult exec(ContextAwareCommandInvocation invocation) {
+      if (help) {
+         invocation.println(invocation.getHelpInfo());
+         return CommandResult.SUCCESS;
+      }
+      if (version) {
+         invocation.printf("%s CLI %s\n", org.infinispan.commons.util.Version.getBrandName(), org.infinispan.commons.util.Version.getBrandVersion());
+         invocation.printf("Copyright (C) Red Hat Inc. and/or its affiliates and other contributors\n");
+         invocation.printf("License Apache License, v. 2.0. http://www.apache.org/licenses/LICENSE-2.0\n");
+         return CommandResult.SUCCESS;
+      }
+
+      context = invocation.getContext();
+
+      if (truststore != null) {
+         try (FileInputStream f = new FileInputStream(truststore.getAbsolutePath())) {
+            KeyStore keyStore = KeyStoreUtil.loadKeyStore(ProviderUtil.INSTALLED_PROVIDERS, null, f, truststore.getAbsolutePath(), truststorePassword.toCharArray());
+            TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            trustManagerFactory.init(keyStore);
+            SSLContextSettings sslContext = SSLContextSettings.getInstance("TLS", null, trustManagerFactory.getTrustManagers(), null, null);
+            context.setSslContext(sslContext);
+         } catch (Exception e) {
+            invocation.getShell().writeln(MSG.keyStoreError(truststore.getAbsolutePath(), e));
+            return CommandResult.FAILURE;
+         }
+      } else if (trustAll) {
+         SSLContextSettings sslContext = SSLContextSettings.getInstance("TLS", null, new TrustManager[]{new ZeroSecurityTrustManager()}, null, new ZeroSecurityHostnameVerifier());
+         context.setSslContext(sslContext);
+      }
+
+      if (connect != null) {
+         context.connect(null, connect);
```

Comment:
suggestion: add options to set the user/password so users could do:
`./bin/cli.sh -c - -u <user> -p <pass> [-f commands.cli]`
wdyt?

Reply:
Yet one more way to do `cli.sh -c http://user:pass@localhost:11222` ?